177190907 investigate meeseeks crash

dushyantss opened the pull request.

dushyantss committed:
[177190907]:FEAT Update response and handle invalid html

* Also allow sending a default icon url

dushyantss committed:
[177190907]:FIX Handle case of meeseeks error

neilberkman reviewed (changes requested):
A test emulating the issue with Paul Graham article would be nice. All of the special cases we're finding regarding article content have high potential for regressions as we go along and keeping comprehensive tests will help avoid bug "whack-a-mole".


![](https://gifimage.net/wp-content/uploads/2018/06/whack-a-mole-gif-3.gif)

dushyantss committed:
[177190907]:FIX Add tests for edge cases

* Handle paulgraham.com
* Add an attempt to encode the returned html to latin1, in hopes that we may parse it.

dushyantss commented:
Here are the tests for paulgraham.com. Also added a few more tests, added one more attempt to encode the returned string and cleaned up the code. Also, sorry about the potential whack-a-mole 😓.

neilberkman reviewed (approved):
Great! 🎉
